### DISCUSSION TITLE
Does the multi-track plugin support video?

### DISCUSSION THREAD

**Discussion by MikeSamy** [katspaugh/wavesurfer.js#3133] [Q&A]
I'm setting up the tracks the same way I would in this demo - https://wavesurfer-js.org/examples/?video.js, using: 

`media: document.querySelector('video')`

in place of 

`url: 'file.mp3'`

but the tracks are empty upon loading. Does the multi-track plugin support this way of loading media? 

I'm trying to make a player that plays multiple, scrubbable videos at the same time.

**Comment by katspaugh** [comment]
It doesn't support custom audio or video elements out of the box, because it creates its own audio elements under the hood.

I don't remember now why I did it this way. It should be possible to refactor multitrack to not do that and just use wavesurfer instances.

**Comment by glassworks-projects** [comment]
I am looking to implement a player that plays video and audio in sync, so I can take a stab at this refactor, except I can't find the Multitrack source code anywhere.

**Reply by katspaugh** [reply]
It's here: https://github.com/katspaugh/wavesurfer-multitrack
